interfaces/builtin/service_watchdog.go

```diff
@@ -53,15 +53,35 @@ func (iface *serviceWatchdogInterface) AppArmorConnectedPlug(spec *apparmor.Spec
 	notifySocket := osGetenv("NOTIFY_SOCKET")
 	if notifySocket == "" {
 		notifySocket = "/run/systemd/notify"
-	} else {
-		// must be an absolute path without any AppArmor regular
-		// expression (AARE) characters or double quotes
-		if !strings.HasPrefix(notifySocket, "/") || strings.ContainsAny(notifySocket, `?*[]{}^"`) {
-			return fmt.Errorf("cannot use %q as notify socket path", notifySocket)
-		}
 	}
+
+	if !(strings.HasPrefix(notifySocket, "/") || strings.HasPrefix(notifySocket, "@")) || strings.ContainsAny(notifySocket, `?*[]{}^"`) {
+		// must be an absolute path or an abstract socket path, without
+		// any AppArmor regular expression (AARE) characters or double
+		// quotes
+		return fmt.Errorf("cannot use %q as notify socket path", notifySocket)
+	}
+
+	var rule string
+
+	switch {
+	case strings.HasPrefix(notifySocket, "/"):
+		rule = fmt.Sprintf(`"%s" w`, notifySocket)
+	case strings.HasPrefix(notifySocket, "@/org/freedesktop/systemd1/notify/"):
+		// special case for Ubuntu 14.04 where the manpage states that
+		// /run/systemd/notify is used, but in fact the services get an
+		// abstract socket path such as
+		// @/org/freedesktop/systemd1/notify/13334051644891137417, the
+		// last part changing with each reboot
```

Comment:
This has the interesting side-effect that on 14.04, every snap that uses this interface will unconditionally have its profile recompiled on every boot. Is this worth noting in this comment? (add at your discretion)

Reply:
As discussed on IRC the profile is not recompiled. NOTIFY_SOCKET changes on each reboot, but we have a catchall rule that covers it.